Pcap++/header/PcapFilter.h

```diff
@@ -243,62 +243,145 @@ namespace pcpp
+		 * @param[in] dir The address direction to filter (source or destination)
+		 * @throws std::invalid_argument The provided address is not a valid IPv4 or IPv6 address.
+		 */
+		IPFilter(const std::string& ipAddress, Direction dir) : IPFilter(IPAddress(ipAddress), dir) {}
+
+		/**
+		 * The basic constructor that creates the filter from an IP address and direction (source or destination)
+		 * @param[in] ipAddress The IP address to build the filter with.
 		 * @param[in] dir The address direction to filter (source or destination)
 		 */
-		IPFilter(const std::string& ipAddress, Direction dir) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(""), m_Len(0) {}
+		IPFilter(const IPAddress& ipAddress, Direction dir) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(""), m_Len(0) {}
 
 		/**
 		 * A constructor that enable to filter only part of the address by using a mask (aka subnet). For example: "filter only IP addresses that matches
 		 * the subnet 10.0.0.x"
 		 * @param[in] ipAddress The IPv4 address to use. Only the part of the address that is not masked will be matched. For example: if the address
-		 * is "1.2.3.4" and the mask is "255.255.255.0" than the part of the address that will be matched is "1.2.3.X". If this address is not a
-		 * valid IPv4 address an error will be written to log and parsing this filter will fail
+		 * is "1.2.3.4" and the mask is "255.255.255.0" than the part of the address that will be matched is "1.2.3.X".
 		 * @param[in] dir The address direction to filter (source or destination)
 		 * @param[in] ipv4Mask The mask to use. Mask should also be in a valid IPv4 format (i.e x.x.x.x), otherwise parsing this filter will fail
+		 * @throws std::invalid_argument The provided address is not a valid IPv4 address.
+		 */
+		IPFilter(const std::string& ipAddress, Direction dir, const std::string& ipv4Mask) : IPFilter(IPv4Address(ipAddress), dir, ipv4Mask) {}
+
+		/**
+		 * A constructor that enable to filter only part of the address by using a mask (aka subnet). For example:
+		 * "filter only IP addresses that matches the subnet 10.0.0.x"
+		 * @param[in] ipAddress The IPv4 address to use. Only the part of the address that is not masked will be
+		 * matched. For example: if the address is "1.2.3.4" and the mask is "255.255.255.0" than the part of the
+		 * address that will be matched is "1.2.3.X".
+		 * @param[in] dir The address direction to filter (source or destination)
+		 * @param[in] ipv4Mask The mask to use. Mask should also be in a valid IPv4 format (i.e x.x.x.x), otherwise
+		 * parsing this filter will fail
 		 */
-		IPFilter(const std::string& ipAddress, Direction dir, const std::string& ipv4Mask) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(ipv4Mask), m_Len(0) {}
+		IPFilter(const IPv4Address& ipAddress, Direction dir, const std::string& ipv4Mask) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(ipv4Mask), m_Len(0) {}
```

Comment:
if the IP address is already `IPv4Address`, why not let the mask also be `IPv4Address`?

Reply:
Hmm, possibly can be done. 

A couple of things to note tho:
- After the change to use `IPNetwork`, the `IPNetwork` ctor uses `string` for netmask too, so that has to be changed too.
- Changing `string` to `IPAddress` will break any C-strings that are currently used as parameters, as it becomes a double conversion (`C-string`->`string`->`IPAddress`) so the new ctors can be added but changing the type directly will break backwards compat.

Reply:
I think we can keep it as is